exo/inference/mlx/sharded_utils.py

```diff
@@ -107,17 +105,14 @@ def load_model_shard(
         raise FileNotFoundError(f"No safetensors found in {model_path}")
 
     weights = {}
-    all_weights_keys = set()
     for wf in weight_files:
-        weights_dict = mx.load(wf)
-        all_weights_keys.update(weights_dict.keys())
-        weights.update({k: v for k, v in weights_dict.items() if not k.startswith("model.layers.") or shard.start_layer <= int(k.split('.')[2]) <= shard.end_layer})
+        weights.update(mx.load(wf))
```

Comment:
Isn't this going to load the entire model into memory? We need to make sure only the layers for this shard are loaded into memory.

Reply:
I don't know the exact details of `mx.load` implementation, but it should be lazy evaluation. That's why when you use mlx_lm.convert, you can convert a large model (>700GB) without requiring that amount of system memory. From my quick test, I don't feel it loads the entire weight into memory. Perhaps @awni could help confirm this.

Reply:
@AlexCheema based on the clarification from the MLX team, the above changes should be fine..detials -> https://github.com/ml-explore/mlx/discussions/1292

Reply:
Awesome - thanks for getting clarification on that